File: eform-client/src/app/plugins/modules/backend-configuration-pn/models/task-wizard/task-wizard-task.model.ts
import {RepeatTypeEnum, TaskWizardStatusesEnum} from '../../enums';
import {CommonTranslationsModel} from 'src/app/common/models';

export interface TaskWizardTaskModel {
  id: number;
  propertyId: number;
  folderId: number;
  itemPlanningTagId: number,
  tags: number[];
  translations: CommonTranslationsModel[];
  eformId: number;
  eformName: string;
  startDate: Date;
  repeatType: RepeatTypeEnum;
  repeatEvery: number;
  status: TaskWizardStatusesEnum;
  assignedTo: number[];
}

File: eFormAPI/Plugins/BackendConfiguration.Pn/BackendConfiguration.Pn/Infrastructure/Models/TaskWizard/TaskWizardTaskModel.cs
namespace BackendConfiguration.Pn.Infrastructure.Models.TaskWizard;

using System;
using Enums;
using Microting.eFormApi.BasePn.Infrastructure.Models.Common;
using System.Collections.Generic;

public class TaskWizardTaskModel
{
    public int Id { get; set; }
    public int PropertyId { get; set; }
    public int FolderId { get; set; }
    public int? ItemPlanningTagId { get; set; }
    public List<int> Tags { get; set; } = [];
    public List<CommonTranslationsModel> Translations { get; set; } = [];
    public int EformId { get; set; }
    public string EformName { get; set; }
    public DateTime StartDate { get; set; }
    public RepeatType RepeatType { get; set; }
    public int RepeatEvery { get; set; }
    public TaskWizardStatuses Status { get; set; }
    public List<int> AssignedTo { get; set; } = [];
    public bool ComplianceEnabled { get; set; }
}